packages/w3up-client/src/client.js

```diff
@@ -282,6 +282,76 @@ export class Client extends Base {
     return space
   }
 
+  /**
+   * Share an existing space with another Storacha account via email address delegate.
+   * Delegates access to the space to the specified email account with the following permissions:
+   * - space/* - for uploading blobs
+   * - upload/*- for registering uploads
+   * - filecoin/* - for submitting to the filecoin pipeline
+   * - access/* - for re-delegating access to other devices
+   * - usage/* - for querying usage
+   * The default expiration is set to infinity.
+   *
+   * @typedef {Omit<import('./types.js').UCANOptions, 'audience'> & { audienceMeta?: import('./types.js').AgentMeta }} ShareOptions
+   *
+   * @param {import("./types.js").EmailAddress} delegateEmail - Email of the account to share the space with.
+   * @param {import('./types.js').SpaceDID} spaceDID - The DID of the space to share.
+   * @param {import('./types.js').ServiceAbility[]} abilities - Abilities to delegate to the delegate account.
+   * @param {ShareOptions} [options] - Options for the delegation.
+   *
+   * @returns {Promise<import('./delegation.js').AgentDelegation<any>>} Resolves with the AgentDelegation instance once the space is successfully shared.
+   * @throws {Error} - Throws an error if there is an issue delegating access to the space.
+   */
+  async shareSpace(
+    delegateEmail,
+    spaceDID,
+    abilities = [
+      'space/*',
+      'upload/*',
+      'access/*',
+      'usage/*',
+      'filecoin/offer',
+      'filecoin/submit',
+      'filecoin/accept',
+      'filecoin/info',
+    ],
+    options = { expiration: Infinity }
+  ) {
+    // Make sure the agent is using the space before delegating
+    await this.agent.setCurrentSpace(spaceDID)
```

Comment:
Urgh, this is why we need a client refactor...current space was such a bad idea.

This should probably do one of three things:

1. Reset current space if different to sharing space
2. Allow a space that isn't the current space to be passed to `agent.delegate`
3. Only operate on the current space

I would vote option `#2`.

Reply:
Agree. Option `#2` makes more sense. Do you think we should tackle it as part of this change or in a new PR? Just wondering if the refactor won't touch several places.

Reply:
I would submit a separate PR and then rebase this one once it is merged.

I know it doesn't matter so much with this PR as it's a feature release but if this one was a breaking change, and you added that feature to access client here, then release please would make a major version for access client as well, when it actually only needed a minor version bump. I think it's good to exercise disipline even if not strictly required. Note: there are other benefits to landing it separately, like cleaner changelog...